Added Animation, Tilemaps, and many more (#24)
twolivesleft/Codea4-Docs: docs/source/api/input.rst

   .. lua:attribute:: touchCount: integer

      The current number of touches associated with this gesture
      
      .. helptext:: get the touch count for this gesture

   .. lua:attribute:: direction: enum

      The direction of the swipe
      

>       .. helptext:: get the direction for this gesture

Reviewer: ```suggestion
      .. helptext:: get the direction of the swipe
```

   .. lua:attribute:: enabled: boolean

      Enables/disables this gesture recognizer
      
      .. helptext:: whether the gesture recognizer is enabled

.. lua:class:: gesture.swipe

   Swipe gesture recognizer (using system gesture recognizer for implementation)

   .. lua:staticmethod:: gesture.swipe(callback[, swipeDirection = gesture.all, touchCount = 1])

      Creates and registers a new swipe gesture recognizer that will call ``callback(gesture)`` when recognized

>       :return: The gestures in this order (left, right, up, down). But if a direction is not included then it is ingored

Reviewer: ```suggestion
      :return: The gestures in this order (left, right, up, down). But if a direction is not included, it is ignored.
```

      :rtype: gesture.swipe, gesture.swipe, gesture.swipe, gesture.swipe 
      
      .. helptext:: create a swipe gesture recognizer

   .. lua:attribute:: enabled: boolean

>       Enables/disables this gesture recognizer

Reviewer: Missing helptext.

      
      .. helptext:: whether the gesture recognizer is enabled

.. lua:class:: gesture.longPress

   Rotation gesture recognizer (using system gesture recognizer for implementation)

   .. lua:staticmethod:: gesture.longPress(callback[, tapCount = 0, touchCount = 1, allowableMovement = 10, minimumPressDuration = 0.5])
      
      Creates and registers a new long press gesture recognizer that will call ``callback(gesture)`` when recognized
      
      .. helptext:: create a long press gesture recognizer

   .. lua:attribute:: enabled: boolean

      Enables/disables this gesture recognizer
      

>       .. helptext:: whether the gesture recognizer is enabled

Reviewer: ```suggestion
      .. helptext:: get or set if the gesture recognizer is enabled
```

Reviewer: Look for other places where you have "whether the gesture recognizer is enabled" and standardize to "get or set if the gesture recognizer is enabled"

         .. helptext:: get if the directional pad is moved up

      .. lua:attribute:: down: boolean
      
         .. helptext:: get if the directional pad is moved down

Mouse
########

.. lua:currentmodule:: None

.. lua:class:: mouse

   .. lua:attribute:: active: boolean

>       Is there a mouse active

Reviewer: ```suggestion
      Checks if a mouse is currently active.
```

      
      .. helptext:: checks if a mouse is active

   .. lua:attribute:: connected: function(mouse)

      Callback for when a mouse is connected
      

>       .. helptext:: sets a callback to call when a mouse is connected

Reviewer: ```suggestion
      .. helptext:: callback to call when a mouse is connected
```

   .. lua:attribute:: disconnected: function(mouse)

      Callback for when a mouse is disconnected
      

>       .. helptext:: sets a callback to call when a mouse is disconnected

Reviewer: ```suggestion
      .. helptext:: callback to call when a mouse is disconnected
```

   
      .. helptext:: gets the delta X of the mouse
   
   .. lua:attribute:: dy: number
   
      .. helptext:: gets the delta Y of the mouse

   .. lua:attribute:: deltaX: number
   
      .. helptext:: gets the delta X of the mouse
   
   .. lua:attribute:: deltaY: number
   
      .. helptext:: gets the delta Y of the mouse

   .. lua:attribute:: delta: vec2

      Return a vec2 of both dx and dy
      
      .. helptext:: gets the delta of the mouse

   .. lua:attribute:: visible: boolean

      Sets whether the mouse is visible or hidden
      
      .. helptext:: set the visibility of the mouse

   .. lua:class:: button

      .. lua:attribute:: pressing: boolean
      
         .. helptext:: get whether this button is being pressed

      .. lua:attribute:: pressed: 
      
         .. helptext:: get whether this button was just pressed

      .. lua:attribute:: released: boolean
      
         .. helptext:: get whether this button was just released

      .. lua:attribute:: value: number
      
         .. helptext:: get the analog value of this button

      .. lua:attribute:: touching: boolean
      
         .. helptext:: get whether the touchpad is being touched

> .. lua:module:: mouse

Reviewer: Does this actually exist yet in Codea? If not, we shouldn't document it.
